content-services/content-sources-backend
# Fixes 2941: Add listSnapshotErrata to api

## Summary

- Adds an endpoint to list snapshot errata: `GET /snapshots/{uuid}/errata`
- Support for filtering on name / synopsis, type, severity
- Dependent on this [tang PR](https://github.com/content-services/tang/pull/6)

## Testing steps

- Add a repository with package errata (https://dl.fedoraproject.org/pub/epel/7/x86_64/ is a good one but takes a bit of time) and let it snapshot
- Make a request, will look something like this:
`GET /snapshots/597779df-e5f9-4db1-97e0-ced3074c8beb/errata`
- Can also make a request that includes optional filtering, which will look something like this: `GET /snapshots/597779df-e5f9-4db1-97e0-ced3074c8beb/errata?type=security&severity=Critical&search=epel`
- You can filter on types `security`, `enhancement`, `bugfix`, and `other`
- You can filter on severities `Important`, `Critical`, `Moderate`, `Low`, and `Unknown`
- You can filter on multiple types and severities (`/snapshots/:uuid/errata?type=security,enhancement&severity=Critical,Moderate`)

- Should return an array of results, here's one example:

```
[
     {
            "id": "018ec320-5315-7174-ae8b-fbeff290a222",
            "errata_id": "FEDORA-EPEL-2023-6569f44fa5",
            "title": "moin-1.9.11-1.el7",
            "summary": "moin-1.9.11-1.el7 security update",
            "description": "Security update",
            "issued_date": "2023-01-06 17:28:59",
            "updated_date": "2023-01-15 00:45:13",
            "type": "security",
            "severity": "Critical",
            "reboot_suggested": false
     },
     {
            "id": "018ec31f-a55f-7cf4-b908-1b5f4c2a0384",
            "errata_id": "FEDORA-EPEL-2022-db09048bde",
            "title": "nbd-3.24-1.el7",
            "summary": "nbd-3.24-1.el7 security update",
            "description": "Update to 3.24: fix CVE-2022-26495, CVE-2022-26496\n",
            "issued_date": "2022-03-08 02:53:28",
            "updated_date": "2022-03-16 15:31:09",
            "type": "security",
            "severity": "Critical",
            "reboot_suggested": false
        },
]     
```

## Checklist
